specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/preview/0.1-preview/confidentialledger.json

```diff
@@ -0,0 +1,858 @@
+      "description": "An entry in the ledger.",
+      "required": [
+        "contents"
+      ],
+      "properties": {
+        "contents": {
+          "type": "string",
+          "description": "Contents of the ledger entry."
+        },
+        "subLedgerId": {
+          "$ref": "#/definitions/SubLedgerId",
+          "readOnly": true
+        },
+        "transactionId": {
+          "$ref": "#/definitions/TransactionId",
+          "readOnly": true,
+          "description": "The transaction ID identifying the point in time in which this ledger entry was read."
+        }
+      },
+      "type": "object"
+    },
+    "LedgerQueryResult": {
+      "description": "The result of querying for a ledger entry from an older transaction id. The ledger entry is available in the response only if the returned state is Ready.",
+      "required": [
+        "state"
+      ],
+      "properties": {
+        "state": {
+          "$ref": "#/definitions/LedgerQueryState"
+        },
+        "entry": {
+          "description": "The ledger entry found as a result of the query. This is only available if the query is in Ready state.",
+          "$ref": "#/definitions/LedgerEntry"
+        }
+      },
+      "type": "object"
+    },
+    "LedgerQueryState": {
+      "description": "State of a ledger query.",
+      "type": "string",
+      "enum": [
+        "Loading",
+        "Ready"
+      ],
+      "x-ms-enum": {
+        "name": "ConfidentialLedgerQueryState",
+        "modelAsString": true
+      }
+    },
+    "LedgerUser": {
+      "description": "Details about a Confidential Ledger user.",
+      "required": [
+        "assignedRole"
+      ],
+      "properties": {
+        "assignedRole": {
+          "$ref": "#/definitions/LedgerUserRole"
+        },
+        "userId": {
+          "$ref": "#/definitions/UserId",
+          "readOnly": true
+        }
+      },
+      "type": "object"
+    },
+    "LedgerUserRole": {
+      "type": "string",
+      "description": "Represents an assignable role.",
+      "enum": [
+        "Administrator",
+        "Contributor",
+        "Reader"
+      ],
+      "x-ms-enum": {
+        "name": "ConfidentialLedgerUserRoleName",
+        "modelAsString": true
+      }
+    },
+    "LedgerWriteResult": {
+      "description": "Returned as a result of a write to the Confidential Ledger, the transaction id in the response indicates when the write will become durable.",
+      "required": [
+        "subLedgerId"
+      ],
+      "properties": {
+        "subLedgerId": {
+          "$ref": "#/definitions/SubLedgerId"
+        }
+      },
+      "type": "object"
+    },
+    "MerkleProof": {
+      "description": "The Merkle proof verifying a transaction.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/MerkleProofElement"
+      }
+    },
+    "MerkleProofElement": {
+      "description": "An item in the Merkle proof.",
+      "type": "object",
+      "properties": {
```

Comment:
In the previous PR, these were required. Was that an intentional change?

Reply:
That's correct, there were some underlying technical changes that required this.